src/client/clientenvironment.cpp

```diff
@@ -404,12 +401,12 @@ void ClientEnvironment::addActiveObject(u16 id, u8 type,
 			<<std::endl;
 	}
 
-	u16 new_id = addActiveObject(obj);
+	u16 new_id = addActiveObject(std::move(obj));
```

Comment:
i checked this function signature, i think we can diverge from ServerEnvironment, as there is no inheritance.
this will prevent the just next line getter. Can se return directly the CAO pointer instead (nullptr if register failed) ?

Reply:
by preventing that we will improve performance if there is huge amount of entities, removing one lookup

Reply:
Sure, that would be possible.
But I'd not like to stuff more changes into this PR.